context/grpc/src/main/java/brave/context/grpc/GrpcCurrentTraceContext.java

```diff
@@ -0,0 +1,26 @@
+package brave.context.grpc;
+
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.TraceContext;
+import io.grpc.Context;
+
+/** Manages scope with a {@link Context} as opposed to a thread local. */
+public final class GrpcCurrentTraceContext extends CurrentTraceContext {
```

Comment:
@kuroneko25 can you copy/paste this file into your project, then wrap it with the log4j2 context and see if it works for you?

ex. `ThreadContextCurrentTraceContext.create(GrpcCurrentTraceContext.create())`